framework/src/bundle/BundleManifest.cpp

```diff
@@ -143,34 +177,41 @@ bool BundleManifest::Contains(const std::string& key) const
 Any BundleManifest::GetValue(const std::string& key) const
 {
   auto iter = m_Headers.find(key);
-  if (iter != m_Headers.end()) {
+  if (m_Headers.cend() != iter)
+  {
     return iter->second;
   }
   return Any();
 }
 
 Any BundleManifest::GetValueDeprecated(const std::string& key) const
 {
+  copy_deprecated_properties(m_Headers, m_PropertiesDeprecated);
   auto iter = m_PropertiesDeprecated.find(key);
-  if (iter != m_PropertiesDeprecated.end()) {
+  if (m_PropertiesDeprecated.cend() != iter)
+  {
     return iter->second;
   }
   return Any();
 }
 
 std::vector<std::string> BundleManifest::GetKeysDeprecated() const
 {
+  copy_deprecated_properties(m_Headers, m_PropertiesDeprecated);
   std::vector<std::string> keys;
-  for (AnyMap::const_iterator iter = m_PropertiesDeprecated.begin();
-       iter != m_PropertiesDeprecated.end();
-       ++iter) {
+  for (AnyMap::const_iterator iter = m_PropertiesDeprecated.cbegin();
+       iter != m_PropertiesDeprecated.cend();
+       ++iter)
+  {
     keys.push_back(iter->first);
   }
   return keys;
 }
 
 std::map<std::string, Any> BundleManifest::GetPropertiesDeprecated() const
 {
+  copy_deprecated_properties(m_Headers, m_PropertiesDeprecated);
```

Comment:
there is a possible race condition if more than one thread calls a deprecated bundle property API. I'd suggest using ```std::call_once``` for all the calls to ```copy_deprecated_properties```

Reply:
Done.